src/query/atomWithQuery.ts

```diff
@@ -88,119 +82,24 @@ export function atomWithQuery<
 ): WritableAtom<TData | undefined, AtomWithQueryAction, void | Promise<void>> {
-  type Result = QueryObserverResult<TData, TError>
-
-  // HACK: having state in atom creator function is not ideal
-  // because, unlike hooks, it's shared by multiple providers.
-  const previousDataCache = new WeakMap<QueryClient, TData>()
-
-  const queryDataAtom = atom((get) => {
-    const queryClient = getQueryClient(get)
-    const options =
-      typeof createQuery === 'function' ? createQuery(get) : createQuery
-    const observer = new QueryObserver(queryClient, options)
-    const initialResult = observer.getCurrentResult()
-    if (
-      initialResult.data === undefined &&
-      options.keepPreviousData &&
-      previousDataCache.has(queryClient)
-    ) {
-      initialResult.data = previousDataCache.get(queryClient)
-    }
-
-    let resolve: ((result: Result) => void) | null = null
-    const makePending = () =>
-      new Promise<Result>((r) => {
-        resolve = r
-      })
-    const resultAtom = atom<Result | Promise<Result>>(
-      initialResult.data === undefined && options.enabled !== false
-        ? makePending()
-        : initialResult
-    )
-    let setResult: ((result: Result) => void) | null = null
-    const listener = (result: Result) => {
-      if (result.isFetching || (!result.isError && result.data === undefined)) {
-        return
-      }
-      if (!resolve && !setResult) {
-        throw new Error('setting result without mount')
-      }
-      if (resolve) {
-        resolve(result)
-        resolve = null
-      }
-      if (setResult) {
-        setResult(result)
-      }
-      if (result.data !== undefined) {
-        previousDataCache.set(queryClient, result.data)
-      }
-    }
-    let unsubscribe: (() => void) | null = null
-    let timer: Timeout | undefined
-    const startQuery = () => {
-      if (!setResult && unsubscribe) {
-        clearTimeout(timer)
-        unsubscribe()
-        unsubscribe = null
-      }
-      if (options.enabled !== false) {
-        if (setResult) {
-          observer.refetch({ cancelRefetch: true }).then(setResult)
-        } else {
-          unsubscribe = observer.subscribe(listener)
-        }
-      }
-      if (!setResult) {
-        // not mounted yet
-        timer = setTimeout(() => {
-          if (unsubscribe) {
-            unsubscribe()
-            unsubscribe = null
-          }
-        }, 1000)
-      }
-    }
-    startQuery()
-    resultAtom.onMount = (update) => {
-      setResult = update
-      if (unsubscribe) {
-        clearTimeout(timer)
-      } else {
-        startQuery()
-      }
-      return () => {
-        setResult = null
-        if (unsubscribe) {
-          unsubscribe()
-          // FIXME why does this fail?
-          // unsubscribe = null
-        }
-      }
-    }
-    return { resultAtom, makePending, startQuery }
+  const getOptions = (get: Getter) => ({
+    staleTime: 500,
```

Comment:
This is to avoid failing tests with strict effect. this or `retryOnMount: false` should help. otherwise, it will fetch twice in DEV mode, which is actually fine and expected.